frontend/lib/Models/group_endpoint_provider.dart

```diff
@@ -0,0 +1,54 @@
+import 'package:adpv_frontend/DataModels/group_endpoints.dart';
+import 'package:flutter/material.dart';
+
+import '../Repository/AdminRepository/groups_repository.dart';
+
+class GroupEndpointProvider with ChangeNotifier {
+  GroupEndpointsData groupEndpointsData = GroupEndpointsData.empty();
+  GroupsRepository repository = GroupsRepository();
+
+  GroupEndpointProvider(Future<GroupEndpointsData> future) {
+    init(future);
+  }
+
+  void init(Future<GroupEndpointsData> future) {
+    future.then((value) {
+      groupEndpointsData = value;
+      groupEndpointsData.sortFields();
+      notifyListeners();
+    });
+  }
+
+  void updateEndpointState(
+    bool? value,
+    String endpointLabel,
+  ) {
+    if (value != null) {
+      groupEndpointsData.endpoints[endpointLabel]!.isBelongingToGroup = value;
+    }
+    notifyListeners();
```

Comment:
Nie wiem, czy używanie operatora `!` jest tutaj bezpieczne. W tym przypadku jeżeli nie znajdzie endpointa to rzuci wyjątkiem, a nie wiem, czy my tego chcemy. Użyłbym `?`, który zrobi update, kiedy wartość nie będzie nullem

Reply:
tylko jeśli go kliknęliśmy, to on na pewno jest, nie ma możliwości żeby to się wysypało tutaj

Reply:
Z `null'em` nie ma żartów i to że nam się wydaje, że dana sytuacja jest niemożliwa, to czasami za mało. Ja byłbym za tym, żeby używać bezpieczniejszej opcji i nie wywalać aplikacji, jeżeli nie będzie można zmienić zaznaczenia check'a